### ISSUE TITLE
API for creating null-terminated string data variables

### ISSUE THREAD

**Issue by jonpalmisc** [Vector35/binaryninja-api#3703]
Sometimes analysis misses strings, or you're dealing with a format that you know has strings at certain offsets, etc.

There is no API for "create a (null-terminated) string data variable starting at this address". As such, we've reinvented this behavior in a few places in the core (`linearview.cpp:5845`, `tokenizedtextview:1877`, and more). So this is really a two part feature request:

1. Consolidate any areas where we do this into the core into one true implementation.
2. Make an API available so plugins, etc. don't have to re-implement the whole "keep reading until you find a null terminator then construct the appropriate array type" process.

**Comment by jrozner** [comment]
Is it necessary for this to only work for null terminated strings? Languages like Go don't use null terminated strings and it would be nice to be able to have the flexibility to mark bytes from here to here are a string and render it as such.

**Comment by jonpalmisc** [comment]
Nope, ideally this would work for Go-/Rust-style strings, but support for that is tracked in a different issue: https://github.com/Vector35/binaryninja-api/issues/3409.

I only limited this to null-terminated strings because of the code duplication mentioned in the issue, and since we support those currently.
